wemake_python_styleguide/visitors/ast/operators.py

```diff
@@ -225,6 +225,22 @@ def visit_NamedExpr(
         self,
         node: ast.NamedExpr,
     ) -> None:
-        """Disallows walrus ``:=`` operator."""
-        self.add_violation(consistency.WalrusViolation(node))
+        """Disallows walrus ``:=`` operator outside comprehension."""
+        closest = walk.get_closest_parent(
+            node,
+            (
+                ast.ListComp,
+                ast.SetComp,
+                ast.DictComp,
+                ast.GeneratorExp,
+            )
+        )
+
+        if closest is None:
```

Comment:
Let's use early returns 

```suggestion
        if closest is not None:
            return
```